packages/itwinui-react/src/core/ToggleSwitch/ToggleSwitch.tsx

```diff
@@ -24,22 +20,41 @@ type ToggleSwitchProps = {
 } & (
   | {
       /**
-       * Size of the toggle switch.
-       *  @default 'default'
+       * Label for the toggle switch.
        */
-      size?: 'default';
+      label: React.ReactNode;
       /**
-       * Custom icon inside the toggle switch. Shown only when toggle is checked and size is not small.
+       * Passes props to wrapper.
        *
-       * Will override the default checkmark icon.
+       * The wrapper is a `label` element when label is provided, and a `div` otherwise.
+       * The prop type changes accordingly.
        */
-      icon?: React.JSX.Element | null;
+      wrapperProps?: React.ComponentProps<'label'>;
```

Comment:
Not sure that it's useful to have this added complexity around `label` vs `div` props. We can simplify it as just `div`?

Reply:
Can we find the lowest common denominator that accepts any kind of HTML attributes? The type of element shouldn't matter.

Reply:
Also I don't think it's a big issue in practice. I expect the most common usage of this prop to be with an inline object literal, or an implicitly typed object, rather than something that is explicitly typed as `ComponentProps<"label">`.